### PR TITLE
almost finished iteration 5

### PR DESCRIPTION
**Pull request by MacInnes**
Synthesis:

1. How to use distribute responsibility across classes - also that we may not have the most detailed specs, and at times we have to use our own judgement to interpret what is wanted.

2.  Maybe I didn't understand question 1, but I simply worked through each iteration's spec to create tests for my interpretation of each spec.  As I wrote more and more tests, the problem became more clear and I was able to refactor appropriately.

3.  running the ERB file, I think I understood how to create it, but we'll never know since I couldn't make it run.

4.  The tests/class creation, once I understood what was wanted.

### CODE REVIEW

File: lib/activity.rb [hunk 1]
```diff
@@ -1,2 +1,39 @@
...
+
+  def initialize(name, base_cost, cost_per_participant)
+    @name = name
+    @base_cost = base_cost
+    @participants = []
+    @cost_per_participant = cost_per_participant
+  end
+
+  def add_participant(name, amt_paid)
+    participant = {
```
Comment (mikedao) [inline lib/activity.rb]: If you werent constrained by how participants were represented, how would you have designed this?

File: lib/activity.rb [hunk 2]
```diff
@@ -1,2 +1,39 @@
...
+  def total_cost
+    @base_cost + @cost_per_participant * @participants.length
+  end
+
+  def fair_share
+    total_cost / @participants.length
+  end
+
+  def amount_owed
+    output = {}
```
Comment (mikedao) [inline lib/activity.rb]: Perhaps a better enumerable could have been used?

File: lib/reunion.rb [hunk 3]
```diff
@@ -1,2 +1,42 @@
...
+
+  def total_cost
+    @activities.reduce(0) do |total, activity|
+      total += activity.total_cost
+      total
+    end
+  end
+
+  def total_amount_owed_per_participant
+    owed_array = @activities.map do |activity|
```
Comment (mikedao) [inline lib/reunion.rb]: Might want to break this up

File: lib/activity.rb (hunk 1 above)
Comment (MacInnes) [reply lib/activity.rb]: with a separate Participant class

File: lib/activity.rb (hunk 2 above)
Comment (MacInnes) [reply lib/activity.rb]: I did refactor it with reduce after time was up, as soon as I wrote output = {} I knew you'd be disappointed in me

**Pull request closed without merging**